rcl/src/rcl/stdatomic_helper.h

```diff
@@ -0,0 +1,111 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__STDATOMIC_HELPER_H_
+#define RCL__STDATOMIC_HELPER_H_
+
+#if !defined(WIN32)
+
+#if defined(__GNUC__) && __GNUC__ <= 4 && __GNUC_MINOR__ <= 9
+// If GCC and below GCC-4.9, use the compatability header.
+#include "stdatomic_helper/gcc/stdatomic.h"
+#elif defined(__clang__) && defined(__has_feature) && !__has_feature(c_atomic)
+// If Clang and no c_atomics (true for some older versions), use the compatability header.
+#include "stdatomic_helper/gcc/stdatomic.h"
+#else
+#include <stdatomic.h>
+#endif
```

Comment:
I usually omit them unless the endif and if are far apart or part of a nested logic. I can go through and add them though. Always add them is a simpler rule to follow anyways, but I'll still avoid some trivial cases like wrapping `extern "C" {` and some boilerplate that is common like the visibility macros: aa83632